conjur_api/providers/authn_authentication_strategy.py

```diff
@@ -0,0 +1,53 @@
+"""
+AuthnAuthenticationStrategy module
+
+This module holds the AuthnAuthenticationStrategy class
+"""
+
+import logging
+from conjur_api.errors.errors import MissingRequiredParameterException
+from conjur_api.http.endpoints import ConjurEndpoint
+from conjur_api.interface.authentication_strategy_interface import AuthenticationStrategyInterface
+from conjur_api.wrappers.http_wrapper import HttpVerb, invoke_endpoint
+
+# pylint: disable=too-few-public-methods
+class AuthnAuthenticationStrategy(AuthenticationStrategyInterface):
+    """
+    AuthnAuthenticationStrategy class
+
+    This class implement the "authn" strategy of authentication
+    """
+    def __init__(
+            self,
+            url: str,
+            account: str,
+            login_id: str
+    ):
+        self._url = url
+        self._account = account
+        self._login_id = login_id
+
+    async def authenticate(self, api_key: str, ssl_verification_data) -> str:
```

Comment:
As mentioned above, having this method take `CredentialsData` via a `SimpleCredentialsProvider` allows the authentication to be handled here.  

As a note, we may need to add a second/alternative `CredentialsData` type to handle hostname/api key.  Users exchange a username/password for an api key, then authenticate with the username/api key combination.  Hosts authenticate with `host/<hostname>`/api key combination (note the `host` in front of the hostname).